Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp

```diff
@@ -269,3 +298,149 @@ void URRFloatingMovementComponent::StopMovementImmediately()
+    }
+    RVOAvoidanceLockVelocity = Velocity;
+    RVOAvoidanceLockTimeout = InDuration;
+}
+
+void URRFloatingMovementComponent::CalculateRVOAvoidanceVelocity(float InDeltaTime)
+{
+    if (RVOAvoidanceWeight >= 1.f || !RVOAvoidanceManager || !GetOwner())
+    {
+        return;
+    }
+
+    if (GetOwner()->GetLocalRole() != ROLE_Authority)
+    {
+        return;
+    }
+
+#if RAPYUTA_SIM_DEBUG
+    const bool bShowDebug = RVOAvoidanceManager->IsDebugEnabled(RVOAvoidanceUID);
+#endif
+
+    //Adjust velocity only if we're in "Walking" mode. We should also check if we're dazed, being knocked around, maybe off-navmesh, etc.
+    if (!Velocity.IsZero() && IsMovingOnGround() && UpdatedPrimitive)
+    {
+        //See if we're doing a locked avoidance move already, and if so, skip the testing and just do the move.
+        if (RVOAvoidanceLockTimeout > 0.f)
+        {
+            Velocity = RVOAvoidanceLockVelocity;
+#if RAPYUTA_SIM_DEBUG
+            if (bShowDebug)
+            {
+                DrawDebugLine(
+                    GetWorld(), GetActorFeetLocation(), GetActorFeetLocation() + Velocity, FColor::Blue, false, 0.5f, SDPG_MAX);
+            }
+#endif
+        }
+        else
+        {
+            FVector newVelocity = RVOAvoidanceManager->GetAvoidanceVelocityForComponent(this);
+            PostProcessRVOAvoidanceVelocity(newVelocity);
+
+            if (!newVelocity.Equals(Velocity))
+            {
+                //Had to divert course, lock this avoidance move in for a short time. This will make us a VO, so unlocked others will know to avoid us.
+                Velocity = newVelocity;
+                SetRVOAvoidanceVelocityLock(RVOAvoidanceManager->LockTimeAfterAvoid);
+#if RAPYUTA_SIM_DEBUG
+                if (bShowDebug)
+                {
+                    DrawDebugLine(GetWorld(),
+                                  GetActorFeetLocation(),
+                                  GetActorFeetLocation() + Velocity,
+                                  FColor::Red,
+                                  false,
+                                  0.05f,
+                                  SDPG_MAX,
+                                  10.0f);
+                }
+#endif
+            }
+            else
+            {
+                //Although we didn't divert course, our velocity for this frame is decided. We will not reciprocate anything further, so treat as a VO for the remainder of this frame.
+                SetRVOAvoidanceVelocityLock(RVOAvoidanceManager->LockTimeAfterClean);    //10 ms of lock time should be adequate.
+#if RAPYUTA_SIM_DEBUG
+                if (bShowDebug)
+                {
+                    //DrawDebugLine(GetWorld(), GetActorLocation(), GetActorLocation() + Velocity, FColor::Green, false, 0.05f, SDPG_MAX, 10.0f);
+                }
+#endif
+            }
+        }
+        RVOAvoidanceManager->UpdateRVO(this);
+
+        bRVOAvoidanceRecentlyUpdated = true;
+    }
+#if RAPYUTA_SIM_DEBUG
```

Comment:
Do you need both RAPYUTA_SIM_DEBUG and bShowDebug?

Reply:
bShowDebug should actually be named bRVODebugEnabled

Reply:
I will use Duc's suggestion:
const bool bRVODebugEnabled = RAPYUTA_SIM_DEBUG && RVOAvoidanceManager->IsDebugEnabled(RVOAvoidanceUID);
then use only bRVODebugEnabled for checks below